core/src/test/java/jenkins/security/HMACConfidentialKeyTest.java

```diff
@@ -38,4 +38,10 @@ public void loadingExistingKey() {
         }
     }
 
+    @Test
+    public void testTruncatedMacOnNonFips() {
+        HMACConfidentialKey key1 = new HMACConfidentialKey("test", 16);
+        String str = key1.mac("Hello World");
+        assertTrue(str, str.matches("[0-9A-Fa-f]{32}"));
```

Comment:
please do not use assertTrue / assertFalse for any condition that is not **directly** a boolean but a computed boolean.
if it fails you do not get told the reason.
Hamcrest has a [matcher](https://hamcrest.org/JavaHamcrest/javadoc/2.2/org/hamcrest/text/MatchesPattern.html#matchesPattern-java.lang.String-) that will match a string against a pattern